packages/prepare-package/src/index.ts

```diff
@@ -16,4 +20,14 @@ export default async function preparePackage (pkgDir: string) {
     }
     await rimraf(path.join(pkgDir, 'node_modules'))
   }
+
+  const included = new Set(await packlist({ path: pkgDir }))
+
+  await walk(pkgDir, async (err, pathname) => {
+    if (err) throw err
+
+    if (!included.has(pathname)) {
+      await fs.unlink(pkgDir + '/' + pathname)
+    }
+  })
```

Comment:
This solution is not optimal. You read all the files from the directory and then in git-fetcher the files are read again.

It would make more sense to return the list of needed files from preparePackages. And add some new API to cafs that would accept a list of files instead of a directory

currently it accepts a directory and read everything from that directory

https://github.com/pnpm/pnpm/blob/2d0f2ba0f262263407ab9d759e316678b87f4813/packages/git-fetcher/src/index.ts#L26

Reply:
Ah ok, I'll see if I can do that.